### PR TITLE
fix: optional path parameter

### PR DESCRIPTION
**Pull request by dalechyn**
Added new parsing util `parseHonoPath` as `parsePath` was also used for routing declaration and it split the path by `?` which is used for declaring optional path parameters, but was optimistically intended to drop search params

That util might need a better name though.

cc @ggomaeng.

### CODE REVIEW

**Comment by dalechyn** [comment]
converting to draft. image route is broken as it's now treated as path parameter...

**Comment by dalechyn** [comment]
To test it out, `pnpm dev` and go to `/routing` path.

File: src/frog-base.tsx [hunk 1]
```diff
@@ -467,33 +500,6 @@ export class FrogBase<
...
-        imageOptions = defaultImageOptions,
-      } = fromQuery<any>(c.req.query())
-      const image_ = JSON.parse(lz.decompressFromEncodedURIComponent(image))
-      return new ImageResponse(image_, {
-        ...imageOptions,
-        fonts: await parseFonts(fonts),
-        headers: imageOptions?.headers ?? headers,
-      })
-    })
-
```
Comment (dalechyn) [inline src/frog-base.tsx:496]: moved this upper to ensure the routing priority is preserved

https://hono.dev/api/routing#routing-priority

File: src/utils/getImatePaths.ts
Comment (jxom) [inline src/utils/getImatePaths.ts:1]: Would be good to have some tests for these added files. 

Also: `src/utils/getImagePaths.ts`

Comment (dalechyn) [reply src/utils/getImatePaths.ts:1]: oopsies. on it.

**Comment by jxom** [comment]
Just needs a changeset now and we're g!

`pnpm changeset`

**Comment by dalechyn** [comment]
> Just needs a changeset now and we're g!
> 
> `pnpm changeset`

It already has one!

File: .changeset/shiny-years-agree.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"frog": patch
+---
+
+Fixed the optional path parameter parsing that resulted in making the path parameter to be required.
```
Comment (dalechyn) [inline .changeset/shiny-years-agree.md]:
```suggestion
Fixed the optional path parameter parsing that resulted in making the path parameter to be required.
Fixed the `/image` route to be prioritized over optional parameter (`/:param?`).
```

**Pull request merged**